Add option to enable TLS build.

T-J-Teru opened the pull request:
Adds a new flag (--with-tls) to the build-all.sh script that enables TLS support.  As TLS support has not yet been merged into the binutils / GCC repositories then the default is for TLS support to be turned off.

I've tested building the standard dev branches, and the TLS branches, and everything looks fine to me.  Is there any concerns with merging this now?

T-J-Teru committed:
Add option to enable TLS build.

Adds new flags (--tls, --no-tls) to the build-all.sh script that
enables (or disables) TLS support.  As TLS support has not yet been
merged into the binutils / GCC repositories then the default is for TLS
support to be disabled (--no-tls).

ChangeLog:

	* build-all.sh: Add tls command line flags, and set TLS_SUPPORT
	environment variable.
	* build-uclibc.sh: Check TLS_SUPPORT environment variable to
	control thread and tls support features.

anthony-kolesov commented on build-all.sh:
```diff
@@ -48,6 +48,7 @@
 #                  [--disable-werror | --no-disable-werror]
 #                  [--strip | --no-strip]
 #                  [--release-name <release>]
+#                  [--with-tls | --without-tls]
```
I would like this to be `--tls | --no-tls` to maintain consistency with other options. It would be much better to use standart `--enable-*|--disable-*` however that should be done for all options then. Perhaps I should rename options for this release, but as of now, for consistency it should be `--tls|--no-tls`.

anthony-kolesov commented on build-all.sh:
```diff
@@ -546,6 +558,7 @@ case ${opt} in
 	echo "                      [--strip | --no-strip]"
 	echo "                      [--sed-tool <tool>]"
 	echo "                      [--release-name <release>]"
+	echo "                      [--with-tls | --without-tls]"x
```
"x" in the end is a typo.

anthony-kolesov commented on build-all.sh:
```diff
@@ -599,6 +612,13 @@ then
+else
+    TLS_SUPPORT="no"
+fi
+
```
Perhaps, you can have `TLS_SUPPORT="no"` as a default, and then move this `if` into the option parsing `case`, to make things simplier. Or even better:

```
--tls) TLS_SUPPORT=yes ;;
--no-tls) TLS_SUPPORT=no ;;
```

That looks cleaner to me. Most of option handing after the case is there because of cross-option dependies, so we cannot do anything until all options has been parsed (there are some exception, which can be moved to `case` as well).

anthony-kolesov commented on build-all.sh:
```diff
@@ -718,6 +738,10 @@ then
 fi
 export SED
 export RELEASE_NAME
+if [ "x${TLS_SUPPORT}" != "x" ]
```
Is there a reason not to export empty TLS_SUPPORT? build-uclibc.sh compares with "yes" anyway so empty TLS_SUPPORT would behave like TLS_SUPPORT=no. In fact, in current configuration TLS_SUPPORT always is either yes or no, so this condition is always true.

anthony-kolesov commented on build-uclibc.sh:
```diff
@@ -339,13 +344,19 @@ cd "${build_dir_stage1}"
 # also disable building libgomp
+if [ "x${TLS_SUPPORT}" = "xyes" ]
+then
+    thread_flags=""
```
I think setting this to `--enable-threads --enable-tls` would be better, to avoid dependencies on default behaviour of configure.

T-J-Teru commented:
@anthony-kolesov I've updated the head to a new commit that I believe addresses all the points you raised.  Let me know if there are any other issues with this merge.  Thanks.

anthony-kolesov commented:
Thanks!
